pytorch_lightning/trainer/trainer.py

```diff
@@ -289,6 +290,15 @@ def __init__(
                 .. deprecated:: 0.5.0
                     Use `min_nb_epochs` instead. Will remove 0.8.0.
 
+            max_steps (int): Stop training after this number of steps.
+                Example::
+
+                    # default used by the Trainer (disabled)
```

Comment:
describe how this interacts with max or min epochs

Reply:
Done.
Since the training loop is epoch based, does it make sense to add a `min_steps` parameter?
My assumption was that if the user specifies `max_steps` he is interested in running training for that number of steps, regardless of a "minimal" number of steps..

Reply:
I think we need:
min_steps, max_steps, min_epochs, max_epochs.

the min training is the max(min_steps, epoch).

Case A:
min_steps = 10
min_epochs = 2
If each epoch has 11 steps, this trains for 2 epochs min.

Case B:
min_steps = 10
min_epochs = 2
If each epoch has 4 steps, then this trains for at least 10 steps.

Case C:
max_steps = 10
max_epochs = 2
if epoch has 11 steps, then this trains for 10 steps.

Case D:
max_steps = 10
max_epochs = 2
if epoch has 2 steps, then this trains for 2 epochs.


Mind making the changes and adding examples in the docs? we can merge after

Reply:
Agreed with the above.
Done.